src/GetApprovedCommentsDynamicFunctionReturnTypeExtension.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Set return type of get_approved_comments() based on its passed arguments.
+ */
+
+declare(strict_types=1);
+
+namespace SzepeViktor\PHPStan\WordPress;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+
+class GetApprovedCommentsDynamicFunctionReturnTypeExtension implements \PHPStan\Type\DynamicFunctionReturnTypeExtension
+{
+    	/**
+	 * @var string[]
+	 */
+	protected static $supported = [
+		'get_approved_comments',
+	];
+
+
+	public function isFunctionSupported( FunctionReflection $functionReflection ): bool {
+		return \in_array( $functionReflection->getName(), static::$supported, true );
+	}
+
+
+	/**
+	 * - Return 'WP_Comment[]' by default.
+	 * - Return `int[]` if `$fields = 'ids'`.
+	 * - Return `int` if `$count = true`.
+	 *
+	 * @link https://developer.wordpress.org/reference/functions/get_approved_comments/#parameters
+	 */
+	public function getTypeFromFunctionCall( FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope ): ?Type {
+		$args = $functionCall->getArgs();
+
+		if ( \count( $args ) < 2 ) {
+			return self::defaultType();
```

Comment:
Isn't is `return null`?

Reply:
I will merge it anyway!

Reply:
https://github.com/szepeviktor/phpstan-wordpress/blob/6fbfbdfc822e70a669cff31d87545fad043f5a55/src/GetTermsDynamicFunctionReturnTypeExtension.php#L64-L66